src/pipeline.ts

```diff
@@ -256,6 +303,74 @@ export class GitHubWorkflow extends PipelineBase {
     return stageDeployment;
   }
 
+  /**
+   * Add a Wave to the pipeline, for deploying multiple Stages in parallel
+   *
+   * Use the return object of this method to deploy multiple stages in parallel.
+   *
+   * Example:
+   *
+   * ```ts
+   * declare const pipeline: GitHubWorkflow; // assign pipeline a value
+   *
+   * const wave = pipeline.addWave('MyWave');
+   * wave.addStage(new MyApplicationStage(this, 'Stage1'));
+   * wave.addStage(new MyApplicationStage(this, 'Stage2'));
+   * ```
+   */
+  public addWave(id: string, options?: WaveOptions): Wave {
+    return this.addGitHubWave(id, options);
+  }
+
+  public addGitHubWave(id: string, options?: WaveOptions): GitHubWave {
+    if (this.builtGH) {
+      throw new Error(
+        "addWave: can't add Waves anymore after buildPipeline() has been called",
+      );
+    }
+
+    const wave = new GitHubWave(id, this, options);
+    this.waves.push(wave);
+    return wave;
+  }
+
+  /**
+   * Support adding stages with GitHub options to waves - should ONLY be called internally.
+   *
+   * Use `pipeline.addWave()` and it'll call this when `wave.addStage()` is called.
+   *
+   * `pipeline.addStage()` will also call this, since it calls `pipeline.addWave().addStage()`.
+   */
+  public addingStageFromWave(
```

Comment:
why is this `adding` rather than `add`?

Reply:
It's an internal call that happens in the middle of the external `addStage()` or `addWave()` - it's the common code for both. I suppose other names could be something like `completeAddingStageFromWave()`.

Reply:
Renamed it to `add` anyway. :-)